### PR TITLE
Feature - RA

### PR DESCRIPTION
**Pull request by joefitter**
* Added router for RA
* Added routes to project router for forking and granting RA
* Fixed billing date test assertion

### CODE REVIEW

File: lib/routers/establishment/projects.js [hunk 1]
```diff
@@ -89,6 +95,28 @@ const submit = action => (req, res, next) => {
...
+  return Promise.resolve()
+    .then(() => RetrospectiveAssessment.query()
+      .select('id', 'status', 'status', 'createdAt', 'updatedAt')
+      .where({ projectId: req.project.id })
+      .orderBy('createdAt', 'desc')
+    )
+    .then(ras => {
+      const draft = ras && ras[0] && ras[0].status === 'draft' ? ras[0] : undefined;
+      // get most recent granted version.
+      const granted = ras.find(v => v.status === 'granted');
```
Comment (lennym) [inline lib/routers/establishment/projects.js:109]: Is it always the case that there is exactly one granted RA?

File: lib/routers/establishment/projects.js [hunk 2]
```diff
@@ -171,7 +199,7 @@ router.param('projectId', (req, res, next, projectId) => {
             .orWhere('additionalEstablishments.id', req.establishment.id);
         })
         .withGraphFetched(
-          '[licenceHolder(constrainParams), collaborators(constrainParams).establishments, establishment(constrainEstablishmentParams), additionalEstablishments(constrainAdditionalEstablishmentParams)]'
+          '[licenceHolder(constrainParams), collaborators(constrainParams).establishments, establishment(constrainEstablishmentParams), additionalEstablishments(constrainAdditionalEstablishmentParams), retrospectiveAssessments]'
```
Comment (lennym) [inline lib/routers/establishment/projects.js:202]: This makes me sad.

Does it work with backticks and newlines?

```
`
[
  licenceHolder(constrainParams),
  collaborators(constrainParams).establishments,
  establishment(constrainEstablishmentParams),
  additionalEstablishments(constrainAdditionalEstablishmentParams),
  retrospectiveAssessments
]
`
```

Comment (joefitter) [reply lib/routers/establishment/projects.js:202]: yeah I know what you mean, I really have no idea why it has to be a string array, and not an array of strings. Backticks should work ok though

File: lib/routers/establishment/projects.js (hunk 1 above)
Comment (joefitter) [reply lib/routers/establishment/projects.js:109]: 👍

File: lib/routers/establishment/project-versions.js [hunk 3]
```diff
@@ -74,7 +74,12 @@ router.param('versionId', (req, res, next, versionId) => {
   Promise.resolve()
     .then(() => ProjectVersion[queryType]()
       .findById(versionId)
-      .withGraphFetched('[project.[licenceHolder(constrainLicenceHolderParams).establishments(constrainEstablishmentParams), establishment(constrainEstablishmentParams)]]')
+      .withGraphFetched(`[
+        project.[
+          licenceHolder(constrainLicenceHolderParams).establishments(constrainEstablishmentParams),
+          establishment(constrainEstablishmentParams)
+        ]
+      ]`)
```
Comment (lennym) [inline lib/routers/establishment/project-versions.js:82]: 🎉

File: lib/routers/establishment/projects.js [hunk 4]
```diff
@@ -160,6 +188,7 @@ router.param('projectId', (req, res, next, projectId) => {
   const { Project } = req.models;
   const { withDeleted } = req.query;
   const queryType = withDeleted ? 'queryWithDeleted' : 'query';
+  console.log('OK THEN');
```
Comment (lennym) [inline lib/routers/establishment/projects.js:191]:
```suggestion
```

Comment (joefitter) [reply lib/routers/establishment/projects.js:191]: :D

**Review by lennym — APPROVED** [review]

**Pull request merged**